examples/wavenet.cc

```diff
@@ -0,0 +1,229 @@
+            where r_in in 0:IN # necessary because r_in gets into unresolved min/max
+       Res0(batch, out)   =       Res0(batch,  out) + NonLin0(batch, out)
+            where out in 0:IN # necessary because r_in gets into unresolved min/max
+
+    Dilate1(batch, out)   =   Bias1(out)
+            where batch in 0:BATCH
+    Dilate1(batch, out)  += Weight1(out, r_in, r_kern) * Data1(batch, r_in, r_kern)
+    NonLin1(batch, out)   = 1 / (1 + exp(-1*(Dilate1(batch, out))))
+    NonLin1(batch, out)  *= tanh(Dilate1(batch, out + 64))
+      Skip1(batch, skip)  =   SkipBias1(skip)
+            where batch in 0:BATCH
+      Skip1(batch, skip) += SkipWeight1(skip, r_in) * NonLin1(batch, r_in)
+            where r_in in 0:IN # necessary because r_in gets into unresolved min/max
+       Res1(batch, out)   =   ResBias1(  out)
+            where batch in 0:BATCH
+       Res1(batch, out)  += ResWeight1(  out, r_in) * NonLin1(batch, r_in)
+            where r_in in 0:IN # necessary because r_in gets into unresolved min/max
+       Res1(batch, out)   =       Res1(batch,  out) + NonLin1(batch, out)
+            where out in 0:IN # necessary because r_in gets into unresolved min/max
+}
+  )TC";
+  tc::ATenCompilationUnit<tc::CudaTcExecutor> atCompl;
+  atCompl.define(tc);
+
+  // 2. Allocate tensors with random data.
+  at::Tensor weight0 = at::CUDA(at::kFloat).rand({128, 64, 2});
+  at::Tensor bias0 = at::CUDA(at::kFloat).rand({128});
+  at::Tensor data0 = at::CUDA(at::kFloat).rand({1, 64, 2});
+  at::Tensor res_weight0 = at::CUDA(at::kFloat).rand({64, 64});
+  at::Tensor res_bias0 = at::CUDA(at::kFloat).rand({64});
+  at::Tensor skip_weight0 = at::CUDA(at::kFloat).rand({256, 64});
+  at::Tensor skip_bias0 = at::CUDA(at::kFloat).rand({256});
+
+  at::Tensor weight1 = at::CUDA(at::kFloat).rand({128, 64, 2});
+  at::Tensor bias1 = at::CUDA(at::kFloat).rand({128});
+  at::Tensor data1 = at::CUDA(at::kFloat).rand({1, 64, 2});
+  at::Tensor res_weight1 = at::CUDA(at::kFloat).rand({64, 64});
+  at::Tensor res_bias1 = at::CUDA(at::kFloat).rand({64});
+  at::Tensor skip_weight1 = at::CUDA(at::kFloat).rand({256, 64});
+  at::Tensor skip_bias1 = at::CUDA(at::kFloat).rand({256});
+
+  // 3. Run autotuning with evolutionary search starting from a naive option.
+  auto naiveOptions = tc::CudaMappingOptions::makeNaiveCudaMappingOptions();
+  tc::autotune::GeneticAutotunerATen geneticAutotuneATen(tc);
+  auto bestOption = geneticAutotuneATen.tune(
+      FLAGS_tuner_proto,
+      "wavenet2layers",
+      {weight0,
+       bias0,
+       data0,
+       res_weight0,
+       res_bias0,
+       skip_weight0,
+       skip_bias0,
+       weight1,
+       bias1,
+       data1,
+       res_weight1,
+       res_bias1,
+       skip_weight1,
+       skip_bias1},
+      naiveOptions);
+
+  // 4. Compile and run the TC with the best option.
+  // Outputs get allocated; could also be pre-allocated and passed.
+  auto handle = atCompl.compile(
+      "wavenet2layers",
+      {weight0,
```

Comment:
let's create a `vector` of tensors once and reuse it instead of copy-pasting code every time